server/routes/snapshotListener.ts

```diff
@@ -0,0 +1,97 @@
+import { Request, Response, NextFunction } from 'express';
+import { USERS } from 'construct-ui/lib/esm/components/icon/generated/IconNames';
+import { Test } from 'mocha';
+import { MixpanelSnapshotEvents } from 'shared/analytics/types';
+import { DB } from '../database';
+import { success } from '../types';
+import SnapshotSpaceCache from '../util/snapshotSpaceCache';
+import { mixpanelTrack } from '../util/mixpanelUtil';
+import { NotificationCategories } from '../../shared/types';
+
+
+const snapshotListener = async (
+  models: DB, cache: SnapshotSpaceCache, req: Request, res: Response, next: NextFunction
+  ) => {
+
+  // Check cache to see if this is a space we care about
+  const validSnapshot = await cache.check(req.body.space);
+
+  if (!validSnapshot) {
+    mixpanelTrack({
+      event: MixpanelSnapshotEvents.SNAPSHOT_INVALID_SPACE,
+      isCustomDomain: false,
+      space: req.body.space
+    });
+    return success(res, "Snapshot POST Recieved, Space not present");
```

Comment:
Is the snapshot firehose expecting any particular sort of response? Or is this arbitrary?

Reply:
Their documentation doesn't mention anything about responses, so I assumed that they were just expecting a 200 and that the message was arbitrary, but this may be worth asking the snapshot people about at some point.